extension/src/Dictionary.ts

```diff
@@ -0,0 +1,165 @@
+      throw new Error(`Could not open file: "${this.dictionaryFile}"`);
+    }
+    const d = JSON.parse(
+      readFileSync(this.dictionaryFile, "utf8")
+    ) as IDictonary;
+    this.language = d.language;
+    this.wordList = d.wordList;
+  }
+
+  toJSON(): unknown {
+    return {
+      language: this.language,
+      wordList: this.wordList,
+    };
+  }
+
+  saveDictionary(): void {
+    const spaces = 2;
+    writeFileSync(
+      this.dictionaryFile,
+      JSON.stringify(this.toJSON(), null, spaces),
+      "utf8"
+    );
+  }
+
+  addWord(word: string, replaceWith: string): Dictionary {
+    if (!this.exists(word)) {
+      const newWord: DictPair = {
+        word: word,
+        replacement: replaceWith,
+        settings: this.defaultSetting(),
+      };
+      this.wordList.push(newWord);
+      this.saveDictionary();
+    }
+    return this;
+  }
+
+  deleteWord(word: string): Dictionary {
+    this.wordList = this.wordList.filter((w) => w.word !== word);
+    this.saveDictionary();
+    return this;
+  }
+
+  find(word: string): DictPair | undefined {
+    return this.wordList.find(
+      (w) => w.word.toLocaleLowerCase() === word.toLocaleLowerCase()
+    );
+  }
+
+  searchAndReplace(text: string): string {
+    this.wordList.forEach((word) => {
+      word.settings = this.defaultSetting(word.settings);
+      const flags = word.settings.matchCasing ? "" : "i";
+      const re = new RegExp(`\\b${word.word}\\b`, flags);
+      const match = text.match(re);
+      if (match) {
+        const replaceValue = word.settings.keepCasingOnFirstCharacter
+          ? Dictionary.keepCasingOnFirstChar(match[0], word.replacement)
+          : word.replacement;
+        text = text.replace(re, replaceValue);
+      }
+    });
+    return text;
+  }
+
+  translateList(wordList: string[]): string[] {
+    return wordList.map((word): string => {
+      return this.translate(word);
+    });
+  }
+
+  translate(word: string): string {
+    const foundWord = this.find(word);
+    if (foundWord === undefined) {
+      return word;
+    }
+    foundWord.settings = this.defaultSetting(foundWord.settings);
+    const translatedWord = foundWord.settings.keepCasingOnFirstCharacter
+      ? Dictionary.keepCasingOnFirstChar(word, foundWord.replacement)
+      : foundWord.replacement;
+    return translatedWord;
+  }
+
+  static keepCasingOnFirstChar(word: string, replacement: string): string {
+    const isUpper = word[0] === word[0].toUpperCase();
+    const firstChar = isUpper
+      ? replacement.charAt(0).toUpperCase()
+      : replacement.charAt(0).toLocaleLowerCase();
+    replacement = `${firstChar}${replacement.slice(1)}`;
+
+    return replacement;
+  }
+
+  exists(word: string): boolean {
+    return this.find(word) !== undefined;
+  }
+
+  defaultSetting(wordSettings?: WordSetting): WordSetting {
+    const defaultSetting: WordSetting = {
+      matchWholeWord: true,
+      matchCasing: true,
+      useRegex: false,
```

Comment:
How is `useRegex` being used, really? :)

Reply:
And `matchWholeWord`? :)

Reply:
and what about `matchWholeWord`? :)

Reply:
Neither of them is used  at the moment. They were mentioned in #152. I can't really see the desired use of these at the moment. Remove?

Reply:
I think they would be great additions to this feature, it's just that it is not yet implemented! 😜

`useRegex: false` should probably mean that the search text should be escaped, otherwise not.
`matchWholeWord : false` should use the boundary (\b) prefix+suffix (as done now), otherwise not.

Agree?

Reply:
Sounds good I'll work it in.